### PR TITLE
Feat: Added Picker component

### PR DESCRIPTION
**Pull request by JoseDuSV**
Summary: Picker implementation for each platform. Will be used for example as nationality selector in passenger section of booking screen.

Related to:
https://github.com/kiwicom/margarita/issues/73
https://github.com/kiwicom/margarita/issues/121

Desktop:
desktop

iOS:
ios

Android:
android

### CODE REVIEW

**Comment by RobinCsl** [comment]
This is in the storybook preview. There might be a small issue. Could you look at it @JoseDuSV ?

screenshot 2019-01-23 at 10 21 00

**Comment by JoseDuSV** [comment]
> This is in the storybook preview. There might be a small issue. Could you look at it @JoseDuSV ?
> 
> screenshot 2019-01-23 at 10 21 00

Thanks, it's fixed now.

**Comment by JoseDuSV** [comment]
@RobinCsl Conflict solved.

And you are right, there is definitely room for improvement in web version. For now there is native select used in passenger form section even on kiwi (only button style is slightly different).

screenshot 2019-01-23 at 14 44 20

But we definitely don't need to strictly follow it and we can make the select items style nicer and closer to class selector on homepage.

screenshot 2019-01-23 at 14 44 45

Can we just merge this PR to make component available and I will create extra issue for web version improvements. Thanks

File: src/Picker/Picker.ios.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+      open: false,
+      selectedValue: pickerValue,
+    });
+    onValueChange(pickerValue);
+  };
+
+  render() {
+    const { open, selectedValue, pickerValue } = this.state;
+    const { optionsData } = this.props;
+    const selectedOption = optionsData.find(
```
Comment (RobinCsl) [inline src/Picker/Picker.ios.js]: This logic, and the corresponding component is duplicated between ios and android. Could you extract it?

File: src/Picker/Picker.ios.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        {open && (
+          <View style={styles.container}>
+            <RNPicker
+              selectedValue={pickerValue}
+              style={styles.picker}
+              onValueChange={this.handlePickerValueChange}
+            >
+              {pickerOptions}
+            </RNPicker>
+            <Button onPress={this.handleOkPress} type="secondary" label="OK" />
```
Comment (RobinCsl) [inline src/Picker/Picker.ios.js]: I am not sure about this label. I guess OK is pretty international. But we really need to be wary of this. What could we do about it?

File: src/Picker/Picker.web.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+import * as React from 'react';
+import { View } from 'react-native';
+import { defaultTokens } from '@kiwicom/orbit-design-tokens';
+import { Icon } from '../Icon';
+import { StyleSheet } from '../PlatformStyleSheet';
+
+import type { Props } from './PickerTypes';
+
+const DEFAULT_OPTION = 'default-empty-option';
```
Comment (RobinCsl) [inline src/Picker/Picker.web.js]: Could you explain what this achieves?

**Comment by RobinCsl** [comment]
Also, this is not strictly related to this PR, but we will need to find a way to properly pass prop types to the components in MDX files as Docz does not parse them if they are outside.
screenshot 2019-01-24 at 12 14 29

**Comment by JoseDuSV** [comment]
> Also, this is not strictly related to this PR, but we will need to find a way to properly pass prop types to the components in MDX files as Docz does not parse them if they are outside.
> screenshot 2019-01-24 at 12 14 29

@RobinCsl  Definitely, I've already created issue for it https://github.com/kiwicom/universal-components/issues/201

File: src/Picker/Picker.ios.js (hunk 1 above)
Comment (JoseDuSV) [reply src/Picker/Picker.ios.js]: Whole duplicate native picker implementation extracted to separate component and duplicate function added to `PickerHelpers`.

File: src/Picker/Picker.ios.js (hunk 2 above)
Comment (JoseDuSV) [reply src/Picker/Picker.ios.js]: Label extracted to props as in other cases.

File: src/Picker/Picker.web.js (hunk 3 above)
Comment (JoseDuSV) [reply src/Picker/Picker.web.js]: Sure. Short description added to const. Hope it will be clearer now.

**Comment by JoseDuSV** [comment]
iOS version now uses modal to show options. So the implementation is more flexible than in first iteration.

picker-ios

File: src/Picker/__tests__/index.test.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      onValueChange={onValueChange}
+      confirmLabel={confirmLabel}
+    />,
+  );
+
+  it('should contain correct label', () => {
+    expect(getByText(testData[1].label)).toBeDefined();
+  });
+
+  it('renders', () => {
```
Comment (RobinCsl) [inline src/Picker/__tests__/index.test.js]: Sometimes I wonder of the usefulness and value of these shallow snapshots. Maybe the previous test is enough to assert that your component at least rendered the right label.

**Pull request merged**